### PR TITLE
Blog App: Create Views

### PR DESCRIPTION
**Pull request by gtekle**
## In this pull request, I have:
- refactored users controller `index` and `show` actions to interact with database
- refactored posts controller `index` and `show` actions to interact with database
- updated `users/index` view to render list of users
- updated `users/show` view to render a given user's details
- updated `posts/index `view to render a give users list of posts
- updated `posts/show` view to render a given post's details

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1,2 +1,38 @@
-<h1>Posts#index</h1>
-<p>Here are list of posts by user <%= params[:user_id]%> </p>
+<div class="user-details">
+  <img src=<%= @user.photo %> alt="<%= @user.name%>'s profile picture" />
+  <div class="user-info">
+    <p class="user_name"><%= @user.name %></p>
+    <p class="user_post-counter">Number of posts: <%= @user.posts_counter %></p>
+  </div>
+</div>
+<ul class="user-posts">
```
Comment (ShoiraTa) [inline app/views/posts/index.html.erb:8]: - [ ] OPTIONAL: Post index page should show all posts written by user. Even though I created 5 posts written by user 1, on the posts index page I can see only 2 posts:
Screenshot 2022-05-03 at 23 08 46

**Review by ShoiraTa — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @gtekle ,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍 
- Project is set up correctly ✔️ 
- You did great by adding validations in views 👏🏻 


parks-and-rec-great-job-gif

## Optional suggestions

- [ ] OPTIONAL: I suggest to update the syntax you wrote the routes:
**From:** 
```
 get "/users/:user_id/posts", to: "posts#index"
  get "/users/:user_id/posts/:id", to: "posts#show"
  get "/users", to: "users#index"
  get "/users/:id", to: "users#show" 
```
**To**: 

```
  resources :users, only: %i[index show] do
    resources :posts, only: %i[index show]
  end
```

As you can see, you code becomes more readable and much shorter 😉 

 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

***
Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gtekle** [comment]
Hi @ShoiraTa 
Thank you very much for the feedback. I will update the routes as per the feedback and I will remove the restriction on the number of posts displayed I put thinking to implement pagination.
:pray:

**Comment by gtekle** [comment]
Hi @ShoiraTa 
I've applied all changes.
:pray:

**Pull request merged**